### ISSUE TITLE
AppSync: Parse "source" and "info" from event as function arguments

### ISSUE THREAD

**Issue by stevewillard** [awslabs/aws-lambda-powertools-python#901]
It would be greatly helpful if the `AppSyncResolver` would also parse the `source` (parent resolved field) and also the `info` (graphql query options) from AppSync resolvers.

**Comment by michaelbrewer** [comment]
@stevewillard yep, that sounds pretty useful. If you have sample events to look at and we can prototype the developer experience for this.

**Comment by heitorlessa** [comment]
@stevewillard bumping to hear what a good experience would look like for you and whether you have a sample event that accompanies that -- we can certainly do it

**Comment by heitorlessa** [comment]
@stevewillard following up on this feature request on whether you can expand on what a good UX would look like

**Comment by stevewillard** [comment]
@heitorlessa Hi! Sorry for the slow follow up.

I actually don't have a good idea for how this would translate to a feature without a major API change. I'm actually quite inexperience with Python, so it's also entirely likely I'm missing something obvious. At the moment I'm doing:
```
parent_id = app.current_event.source["id"]
```

Which is actually pretty effective.

In the past when I have used `apollo-server` (https://www.npmjs.com/package/apollo-server). The resolvers fit into this shape:

```
 resolvers: {
    Query: {
      books: (parent, args, context, info) => {
        console.log(context.myProperty); // Will be `true`!
        return books;
      },
    }
  }
```

Where args are contained into a position argument.

I think the issue is that the arguments for a query are top-level function parameters, as opposed to something contained under something like `args`. I think a naive solution would be to add a named parameter to your resolver function... maybe `appsync_parent` or `appsync_info` to try to avoid collision with arguments 🤔

**Comment by heitorlessa** [comment]
> @heitorlessa Hi! Sorry for the slow follow up.
> 
> 
> 
> I actually don't have a good idea for how this would translate to a feature without a major API change. I'm actually quite inexperience with Python, so it's also entirely likely I'm missing something obvious. At the moment I'm doing:
> 
> ```
> 
> parent_id = app.current_event.source["id"]
> 
> ```
> 
> 
> 
> Which is actually pretty effective.
> 
> 
> 
> In the past when I have used `apollo-server` (https://www.npmjs.com/package/apollo-server). The resolvers fit into this shape:
> 
> 
> 
> ```
> 
>  resolvers: {
> 
>     Query: {
> 
>       books: (parent, args, context, info) => {
> 
>         console.log(context.myProperty); // Will be `true`!
> 
>         return books;
> 
>       },
> 
>     }
> 
>   }
> 
> ```
> 
> 
> 
> Where args are contained into a position argument.
> 
> 
> 
> I think the issue is that the arguments for a query are top-level function parameters, as opposed to something contained under something like `args`. I think a naive solution would be to add a named parameter to your resolver function... maybe `appsync_parent` or `appsync_info` to try to avoid collision with arguments 🤔 
> 
> 

Np at all! Could you share a sample event and schema to help us visualize the problem better?

We can figure out a fluent API later. I just wanna make sure we don't use a hello world event that will lead to a subpar experience for you later

**Comment by heitorlessa** [comment]
Closing due to the lack of response. Please feel free to reopen it if you have more info
